extensions/lifecycle/tasks/LifecycleColdStatusArchiveTask.js

```diff
@@ -0,0 +1,147 @@
+            return done(errors.InternalError
+                .customizeDescription('Unable to obtain client'));
+        }
+
+        const locations = objMD.getLocation();
+        const req = backbeatClient.batchDelete({
+            Locations: locations.map(location => ({
+                key: location.key,
+                dataStoreName: location.dataStoreName,
+                size: location.size,
+                dataStoreVersionId: location.dataStoreVersionId,
+            })),
+            Bucket: bucketName,
+            Key: objectKey,
+            StorageClass: objMD.getDataStoreName(),
+            Tags: JSON.stringify({
+                'scal-delete-marker': 'true',
+                'scal-delete-service': 'lifecycle-transition',
+            }),
+        });
+        return req.send(err => {
+            log.info('action execution ended', entry.getLogInfo());
+
+            if (err && err.statusCode === 404) {
+                log.info('unable to find data to delete',
+                    Object.assign({
+                        method: 'LifecycleColdStatusArchiveTask._executeDeleteData',
+                        bucket: bucketName,
+                        key: objectKey,
+                    }, entry.getLogInfo));
+                return done();
+            }
+
+            if (err) {
+                log.error('an error occurred on deleteData method to backbeat route',
+                    Object.assign({
+                        method: 'LifecycleColdStatusArchiveTask._executeDeleteData',
+                        error: err.message,
+                        httpStatus: err.statusCode,
+                    }, entry.getLogInfo()));
+                return done(err);
+            }
+            return done();
+        });
+    }
+
+    processEntry(entry, done) {
+        const log = this.logger.newRequestLogger();
+        let objectMD;
+        return async.series([
+            next => this._getMetadata(entry, log, (err, res) => {
+                if (err) {
+                    return next(err);
+                }
+
+                objectMD = res;
+                return next();
+            }),
+            next => {
+                const archiveInfo = entry.getArchiveInfo();
+                // set new ObjectMDArchive to ObjectMD
+                objectMD.setArchive(new ObjectMDArchive(archiveInfo));
+                this._putMetadata(entry, objectMD, log, next);
+            },
+            next => {
+                const locationToGC = objectMD.getLocation();
+                if (!locationToGC || (Array.isArray(locationToGC) && locationToGC.length === 0)) {
+                    return process.nextTick(next);
+                }
+
+                return this._executeDeleteData(entry, objectMD, this._log, err => {
+                    if (err) {
+                        return next(err);
+                    }
+
+                    log.debug('successfully deleted location data');
```

Comment:
should we also log some sort of key or id here considering its async it might come out of order and we would have no clue which one was deleted